packages/api/models/rules.js

```diff
@@ -26,25 +27,57 @@ class Rule extends Manager {
   }
 
   async delete(item) {
-    if (item.rule.type === 'scheduled') {
-      const name = `${process.env.stackName}-custom-${item.name}`;
-      await Events.deleteTarget(this.targetId, name);
-      await Events.deleteEvent(name);
+    switch (item.rule.type) {
+      case 'scheduled': {
+        const name = `${process.env.stackName}-custom-${item.name}`;
+        await Events.deleteTarget(this.targetId, name);
+        await Events.deleteEvent(name);
+        break;
+      }
+      case 'kinesis':
+        this.deleteKinesisEventSource(item);
+        break;
+      default:
+        throw new Error('Type not supported');
     }
     return super.delete({ name: item.name });
   }
 
+  /**
+   * update a rule item
+   *
+   * @param {*} original - the original rule
+   * @param {*} updated - key/value fields for update, may not be a complete rule item
+   * @returns {Promise} the response from database updates
+   */
   async update(original, updated) {
     if (updated.state) {
       original.state = updated.state;
     }
 
+    let valueUpdated = false;
     if (updated.rule && updated.rule.value) {
       original.rule.value = updated.rule.value;
+      valueUpdated = true;
     }
 
-    const payload = await Rule.buildPayload(original);
-    await this.addRule(original, payload);
+    switch (original.rule.type) {
+      case 'scheduled': {
+        const payload = await Rule.buildPayload(original);
+        await this.addRule(original, payload);
+        break;
+      }
+      case 'kinesis':
+        if (valueUpdated) {
+          await this.deleteKinesisEventSource(original);
+          await this.addKinesisEventSource(original);
+          updated.rule.arn = original.rule.arn;
+        }
+        else this.updateKinesisEventSource(original);
```

Comment:
Nit: Formatting, according to [style guide](https://github.com/airbnb/javascript#blocks--cuddled-elses) should be 
```javascript
if (valueUpdated) {
  // code
} else {
  this.updateKinesisEventSource(original);
}
```

Reply:
Also, should it have an `await` ?
```javascript
if (valueUpdated) {
  // code
} else {
  await this.updateKinesisEventSource(original);
}
```

Reply:
yes.  updated